frontend/appflowy_tauri/src/appflowy_app/utils/block_change.ts

```diff
@@ -0,0 +1,186 @@
+import { DeltaTypePB } from '@/services/backend/models/flowy-document2';
+import { BlockType, NestedBlock, DocumentState, ChangeType, BlockPBValue } from '../interfaces/document';
+import { Log } from './log';
+import { BLOCK_MAP_NAME, CHILDREN_MAP_NAME, META_NAME } from '../constants/block';
+
+// This is a list of all the possible changes that can happen to document data
+const matchCases = [
+  { match: matchBlockInsert, type: ChangeType.BlockInsert, onMatch: onMatchBlockInsert },
+  { match: matchBlockUpdate, type: ChangeType.BlockUpdate, onMatch: onMatchBlockUpdate },
+  { match: matchBlockDelete, type: ChangeType.BlockDelete, onMatch: onMatchBlockDelete },
+  { match: matchChildrenMapInsert, type: ChangeType.ChildrenMapInsert, onMatch: onMatchChildrenInsert },
+  { match: matchChildrenMapUpdate, type: ChangeType.ChildrenMapUpdate, onMatch: onMatchChildrenUpdate },
+  { match: matchChildrenMapDelete, type: ChangeType.ChildrenMapDelete, onMatch: onMatchChildrenDelete },
+];
+
+export function matchChange(
+  state: DocumentState,
+  {
+    command,
+    path,
+    id,
+    value,
+  }: {
+    command: DeltaTypePB;
+    path: string[];
+    id: string;
+    value: BlockPBValue & string[];
+  },
+  isRemote?: boolean
+) {
+  for (let i = 0; i < matchCases.length; i += 1) {
+    const matchCase = matchCases[i];
+    if (matchCase.match(command, path)) {
+      matchCase.onMatch(state, id, value, isRemote);
+      return;
+    }
+  }
```

Comment:
```suggestion
  const matchCase = matchCases.find((e => e.match(command, path)));
  if (matchCase) {
    matchCase.onMatch(state, id, value, isRemote);
  }
```

Reply:
It looks more simple using `find`

Reply:
If there is only one can response to that command then, I think find is better